src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { bytes_to_hex, Sha256 } from "asmcrypto.js"
+import { TFileInformation, TdrFile } from "./types"
+
+export const getFileInformation: TFileInformation = async (
+  files,
+  progressFunction
+) => {
+  let processedFiles = 1
```

Comment:
I know it was me who said that it would be clearer to report on the number of files processed than the percentage, but now I'm wondering whether we should report on both?

Calculating checksums for big files can be quite slow (see the [results from Alpha](https://github.com/nationalarchives/tdr-dev-documentation/blob/master/technology-considerations/frontend_checksum_options.md#results)), so I think it would be helpful for the progress bar to keep moving as we process a big file.

Reply:
I've changed this now so that the percentage is updated each time a slice of a file is processed. This should give us a reasonably smooth progress bar.